lib/react.dart

```diff
@@ -103,58 +104,204 @@ abstract class Component {
     _jsRedraw();
   }
 
-  /**
-   * set _nextState to newState
-   * and call React original setState method with no parameter
-   */
+  /// Set [_nextState] to provided [newState] value and force a re-render.
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-api.html#replacestate>
   void replaceState(Map newState) {
     Map nextState = newState == null ? {} : new Map.from(newState);
     _nextState = nextState;
+
     _jsRedraw();
   }
 
+  /// ReactJS lifecycle method that is invoked once, both on the client and server, immediately before the initial
+  /// rendering occurs.
+  ///
+  /// If you call [setState] within this method, [render] will see the updated state and will be executed only once
+  /// despite the [state] value change.
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#mounting-componentwillmount>
   void componentWillMount() {}
 
+  /// ReactJS lifecycle method that is invoked once, only on the client _(not on the server)_, immediately after the
+  /// initial rendering occurs.
+  ///
+  /// At this point in the lifecycle, you can access any [ref]s to the children of [rootNode].
+  ///
+  /// The [componentDidMount] method of child [Component]s is invoked _before_ that of parent [Component].
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#mounting-componentdidmount>
   void componentDidMount(/*DOMElement */ rootNode) {}
 
+  /// ReactJS lifecycle method that is invoked when a [Component] is receiving [newProps].
+  ///
+  /// This method is not called for the initial [render].
+  ///
+  /// Use this as an opportunity to react to a prop transition before [render] is called by updating the [state] using
+  /// [setState]. The old props can be accessed via [props].
+  ///
+  /// Calling [setState] within this function will not trigger an additional [render].
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#updating-componentwillreceiveprops>
   void componentWillReceiveProps(newProps) {}
 
+  /// ReactJS lifecycle method that is invoked before rendering when [nextProps] or [nextState] are being received.
+  ///
+  /// This method is not called for the initial [render] or when [redraw] _a.k.a "forceUpdate"_ is used.
```

Comment:
Is this accurate?  since `redraw` isn't exactly the same as the ReactJS `forceUpdate`, I hadn't assumed that `redraw` would circumvent `shouldComponentUpdate`